Snakefile

```diff
@@ -101,7 +104,221 @@ rule multiqc:
     multiqc -o {output} {input} &>> {log}
     """
 
-
+rule fastqc_before_trim_warnings:
+  """
+  Prints warnings for FastQC modules which produced some warnings or fails of the input data.
+  """
+  input:
+    fastqc_report="multiqc_before_trim/multiqc_data/multiqc_fastqc.txt"
+  output:
+    warning_file="WARNING_fastqc_before_trim_overview.txt"
+  log:
+    "logs/fastqc_before_trim_warnings.log"
+  run:
+    with open(input["fastqc_report"], "r") as input_file, open(output["warning_file"], "w") as out_file:
+      fastqc_data = [line.strip() for line in input_file.readlines()[1:]] # skip header
+
+      basic_statistics_list = []
+      per_base_sequence_quality_list = []
+      per_tile_sequence_quality_list = []
+      per_sequence_quality_scores_list = []
+      per_base_sequence_content_list = []
+      per_sequence_gc_content_list = []
+      per_base_n_content_list = []
+      sequence_length_distribution_list = []
+      sequence_duplication_levels_list = []
+      overrepresented_sequences_list = []
+      adapter_content_list = []
+
+      for line in fastqc_data:
+        basic_statistics, per_base_sequence_quality, per_tile_sequence_quality, per_sequence_quality_scores, per_base_sequence_content, per_sequence_gc_content, per_base_n_content, sequence_length_distribution, sequence_duplication_levels, overrepresented_sequences, adapter_content = line.split("\t")[10:]
+        basic_statistics_list.append(basic_statistics)
+        per_base_sequence_quality_list.append(per_base_sequence_quality)
+        per_tile_sequence_quality_list.append(per_tile_sequence_quality)
+        per_sequence_quality_scores_list.append(per_sequence_quality_scores)
+        per_base_sequence_content_list.append(per_base_sequence_content)
+        per_sequence_gc_content_list.append(per_sequence_gc_content)
+        per_base_n_content_list.append(per_base_n_content)
+        sequence_length_distribution_list.append(sequence_length_distribution)
+        sequence_duplication_levels_list.append(sequence_duplication_levels)
+        overrepresented_sequences_list.append(overrepresented_sequences)
+        adapter_content_list.append(adapter_content)
+    
+
+      out_file.write("FASTQC MODULES WHICH PRODUCED SOME WARNINGS OR FAILS\n\n")
+      out_file.write("!!!Please check MultiQC output for interpretation of your data!!!\n")
+      out_file.write('-'*100)
+      out_file.write('\n')
+      # BASIC STATISTICS never produce warnings or fails
+
+      # PER BASE SEQUENCE QUALITY
+      if "warn" in per_base_sequence_quality_list or "fail" in per_base_sequence_quality_list:
+
+        out_file.write("**PER BASE SEQUENCE QUALITY**\n")
```

Comment:
This way of outputting long text line by line is a bit ugly. In Python you can use triple quotes `"""` for multi-line strings.